flink-python/src/main/java/org/apache/flink/python/env/beam/ProcessPythonEnvironmentManager.java

```diff
@@ -326,9 +327,21 @@ private void constructArchivesDirectory(Map<String, String> env) throws IOExcept
 
             // extract archives to archives directory
             for (Map.Entry<String, String> entry : dependencyInfo.getArchives().entrySet()) {
-                ZipUtils.extractZipFileWithPermissions(
-                        entry.getKey(),
-                        String.join(File.separator, archivesDirectory, entry.getValue()));
+                String inFilePath = entry.getKey();
+                String targetDirPath =
+                        String.join(File.separator, archivesDirectory, entry.getValue());
+                if (inFilePath.endsWith(".zip") || inFilePath.endsWith(".jar")) {
```

Comment:
What about convert inFilePath to lower case and then check the suffix?

Reply:
done